channelnotifier/channelnotifier.go

```diff
@@ -0,0 +1,105 @@
+package channelnotifier
+
+import (
+	"sync/atomic"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/subscribe"
+)
+
+// ChannelNotifier is a subsystem which all active, inactive, and closed channel
+// events pipe through. It takes subscriptions for its events, and whenever
+// it receives a new event it notifies its subscribers over the proper channel.
+type ChannelNotifier struct {
+	ntfnServer *subscribe.Server
+
+	started uint32
```

Comment:
atomic vars need to be first on structs, in this case to to ensure they're 32-bit aligned on all architectures

Reply:
hm TIL, fixed